python/ray/tune/trainable.py

```diff
@@ -64,7 +64,9 @@ def __init__(self, config=None, logger_creator=None):
             self._result_logger = logger_creator(self.config)
             self.logdir = self._result_logger.logdir
         else:
-            logdir_prefix = datetime.today().strftime("%Y-%m-%d_%H-%M-%S")
+            logdir_prefix = self._agent_name + "_" + \
```

Comment:
`self._agent_name` and `self._env_id` are not part of the Trainable class, right?

I think this should be done in `agent.py` (or elsewhere) and make it such that the logdir prefix is customizable here.

Reply:
Yes. Those are not part of the Trainable class.
But, from what I can see (please correct me if I'm missing something) currently, there are only two ways a Trainable is used:

1. By Tune
       Whether the Tune API is directly used to run an experiment or a user defined Trainable is registered with the Tune registry, Tune creates a logger when starting a trial and provides it to Trainable and these lines take care of setting the result logger & logdir: https://github.com/ray-project/ray/blob/07cfdf80d2537da3c67dd4f553c5f3e43671cc7d/python/ray/tune/trainable.py#L63-L65

2. By Rllib's Agent interface
     i. If the Rllib agent implementation implements a logger_creator and initializes the Agent class accordingly, the same lines above will handle the result logger & logdir. 
      ii. If the Rllib agent implementation **does not** provide a logger_creator, only then we will hit this line: https://github.com/ray-project/ray/blob/07cfdf80d2537da3c67dd4f553c5f3e43671cc7d/python/ray/tune/trainable.py#L66-L69
So, that's why I thought it will be okay to use the Agent's instance attribute. 

Though this works, I also feel that this doesn't look very good.  The other option I can think of is to create a default tune.Logger object in every Agent instance on construction with the appropriate logdir prefix. This adds another Will that option be okay? What do you guys (@richardliaw , @ericl) think?

Reply:
Hm good catch, if _agent_name isn't an attribute here, then perhaps RLlib agent should always pass in a logger creator to customize the prefix?

Reply:
That sounds fine for customizing the prefix. 
The motivation of this PR was to make the default prefix more informative, I can think of 3 options:
 1. Make the RLlib agents pass in a logger creator by default (instead of the current `None` default)
 2. We could add a new attribute called `_experiment_name` (similar to the existing `_experiment_id`) to Trainable and pass that alone instead of a logger creator
 3. Leave the default as it is

Reply:
Sorry for the late response; let's do option 1.

Thanks!

Reply:
Okay. Done.